code/game/objects/structures/transit_tubes.dm

```diff
@@ -125,17 +126,53 @@ obj/structure/transit_tube_pod/ex_act(severity)
 	..()
 
 
-/obj/structure/transit_tube/station/attack_hand(mob/user as mob)
+/obj/structure/transit_tube/station/attack_hand(mob/user)
 	if(!pod_moving)
 		for(var/obj/structure/transit_tube_pod/pod in loc)
 			if(!pod.moving && pod.dir in directions())
 				if(open)
+					if(!user.lying && user.loc != pod)
+						var/unloaded = 0
+						var/incomplete = FALSE
+
+						for(var/atom/movable/AM in pod)
+							if(isobserver(AM))
+								continue
+							if(unloaded >= TUBE_POD_UNLOAD_LIMIT)
+								incomplete = TRUE
+								break
+							AM.forceMove(get_step(loc, dir))
+							unloaded++
+
+						if(unloaded)
+							user.visible_message("<span class='notice'>[user] unloads [incomplete ? "some things" : "everything"] from the tube pod.</span>", \
+							"<span class='notice'>You unload [incomplete ? "some things" : "everything"] from the tube pod.</span>")
+							return
+
 					close_animation()
 
 				else
 					open_animation()
 
 
+/obj/structure/transit_tube/station/attack_robot(mob/user)
+	if(Adjacent(user))
+		attack_hand(user)
+
+
+/obj/structure/transit_tube_pod/Entered(atom/movable/AM)
+	..()
+	if(ismob(AM) && contents.len > 1)
+		var/mob/M = AM
+		if(M.blinded || !M.client)
+			return
+		var/list/others = contents.Copy() - M
+		for(var/atom/movable/O in others)
+			if(O.invisibility > M.see_invisible)
+				others -= O
+		if(others.len)
+			to_chat(M, "<span class='info'>You notice the tube pod contains [english_list(others)]</span>.")
```

Comment:
Instead of doing it like this, why not just have examining the tube show its contents? They look like they're supposed to be made of glass anyway.

Reply:
My line of thinking was that it would have to be for the individual pod and they tend to layer under the tube, making examining them directly a little awkward. But yeah I could put it in the tube's and the station's Examine too.